src/main/php/PHPMD/Renderer/CheckStyleRenderer.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace PHPMD\Renderer;
+
+use PHPMD\PHPMD;
+use PHPMD\Report;
+
+/**
+ * This class will render a Java-checkstyle compatible xml-report.
+ * for use with cs2pr and others
+ */
+class CheckStyleRenderer extends XMLRenderer
+{
+    /**
+     * Temporary property that holds the name of the last rendered file, it is
+     * used to detect the next processed file.
+     *
+     * @var string
+     */
+    private $fileName;
+
+    /**
+     * Get a violation severity level according to the priority
+     * of the rule that's being broken
+     * @see https://checkstyle.sourceforge.io/version/4.4/property_types.html#severity
+     * - priority 1 maps to error level severity
+     * - priority 2 maps to warning level severity
+     * - priority > 2 maps to info level severity
+     *
+     * @param integer $priority priority of the broken rule
+     * @return string either error, warning or info
+     */
+    protected function mapPriorityToSeverity(int $priority):string
```

Comment:
PHPMD aims to be ran across wide matrix of PHP versions, so the code has to be kept compatible up to PHP 5.4 for PHPMD version 2.x, so scalar type is not possible (it this renderer will need to be on hold until PHPMD version 3.0, and I can't give any ETA for that).

So I recommend:
```suggestion
    protected function mapPriorityToSeverity($priority)
```

Reply:
Fixed in 9e4ab02